sources/rottweiler/src/system.rs

```diff
@@ -0,0 +1,192 @@
+    execute(
+        CRYPTSETUP,
+        &[
+            "luksFormat",
+            "--type",
+            "luks2",
+            "--pbkdf",
+            "pbkdf2",
+            "--pbkdf-force-iterations",
+            "1000",
+            "--batch-mode",
+            device,
+            "-",
+        ],
+        Some(key_data),
+    )?;
+    Ok(())
+}
+
+/// Resize a LUKS device using the provided key
+pub fn cryptsetup_resize(volume_name: &str, key_data: &[u8]) -> Result<()> {
+    execute(
+        CRYPTSETUP,
+        &["resize", "--key-file=-", volume_name],
+        Some(key_data),
+    )?;
+    Ok(())
+}
+
+/// Execute a command with optional stdin input and return stdout
+fn execute(cmd: &str, args: &[&str], input: Option<&[u8]>) -> Result<Zeroizing<Vec<u8>>> {
+    let mut child = Command::new(cmd)
+        .args(args)
+        .stdin(if input.is_some() {
+            Stdio::piped()
+        } else {
+            Stdio::null()
+        })
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn()
+        .with_whatever_context(|_| format!("failed to spawn {}", cmd))?;
+
+    if let Some(data) = input {
+        let mut stdin = child
+            .stdin
+            .take()
+            .with_whatever_context(|| "failed to open stdin")?;
+
+        stdin
+            .write_all(data)
+            .with_whatever_context(|_| "failed to write to stdin")?;
+
+        drop(stdin);
+    }
+
+    let output = child
+        .wait_with_output()
+        .with_whatever_context(|_| format!("failed to wait for {}", cmd))?;
+
+    if !output.status.success() {
+        let stderr = String::from_utf8_lossy(&output.stderr);
+        whatever!(
+            "{} failed with exit code {}: {}",
+            cmd,
+            output.status.code().unwrap_or(-1),
+            stderr.trim()
+        );
+    }
+
+    Ok(Zeroizing::new(output.stdout))
+}
+
+/// Get TPM2 PCRs to bind to based on image features
+///
+/// PCR decoder ring:
+/// - PCR 4: hashes of shim, grub, and vmlinuz
+/// - PCR 7: Secure Boot policy
+/// - PCR 9: kernel command line (includes dm-verity root hash of userspace)
+/// - PCR 11: boot phase (sysinit, preconfigured, configured, ready, shutdown, final)
+/// - PCR 14: machine-owner keys
+///
+/// If in-place updates are disabled, then the measurements in PCR 4 (kernel) and
+/// PCR 9 (userspace) shouldn't change.
+fn get_tpm2_pcrs() -> Result<String> {
+    let features = bottlerocket_image_features::parse_image_features()?;
+    Ok(if features.in_place_updates {
+        "7+11+14".to_string()
+    } else {
+        "4+7+9+11+14".to_string()
+    })
+}
+
+/// Get canonicalized settings from apiclient, excluding random-seed and hostname
+pub fn apiclient_get_settings() -> Result<Zeroizing<Vec<u8>>> {
+    execute(
+        APICLIENT,
+        &[
+            "get",
+            "settings",
+            "--exclude",
+            "settings.updates.random-seed",
```

Comment:
Am I missing an alias? 

```suggestion
            "settings.updates.seed",
```

Reply:
Seems this was just fixed 👍

Reply:
> Am I missing an alias?

No, this was a bug. Too many random numbers on my mind 😀